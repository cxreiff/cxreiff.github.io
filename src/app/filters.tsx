import { FC } from 'react'

export const Filters: FC = () => {
    return (
        <>
            <Turbulence />
        </>
    )
}

const Turbulence = () => (
    <svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="0" height="0">
        <defs>
            <filter id="turbulence_0">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="0" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_1">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="1" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_2">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="2" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_3">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="3" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_4">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="4" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_5">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="5" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_6">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="6" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_7">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="7" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
        </defs> 
    </svg>
)
