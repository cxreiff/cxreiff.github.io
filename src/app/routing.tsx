import { FC, ReactNode, Suspense, useEffect, lazy } from "react";
import { Routes, Route, useLocation } from "react-router-dom";

import { Loader } from "src/common/loader/loader.component";
import { IFrame } from "src/common/iframe/iframe.component";
import {
  DOCUMENT_TITLE_ROOT,
  BREAKOUT_URL,
  PLATFORMER_URL,
  HEX_TILES_URL,
  PIXIJS_URL,
  BABYLONJS_URL,
  PLAIN_WEBGL_URL,
  BIGFOOT_MAP_URL,
} from "src/utilities/constants";

import Homepage from "src/features/homepage/homepage.component";
import Projects from "src/features/projects/projects.component";
import Photos from "src/features/photos/photos.component";
import Sketches from "src/features/sketches/sketches.component";
import ProjectsFigrid from "src/features/projects/projectsFigrid.component";

// const Posts = lazy(() => import("src/features/posts/posts.component"));
// const PostsContent = lazy(
//   () => import("src/features/posts/postsContent.component")
// );

const Shaders = lazy(() => import("src/features/shaders/shaders.component"));

const Pixijs = () => (
  <IFrame aspect={15.5 / 9} url={PIXIJS_URL} title={"pixijs"} canFullscreen />
);
const Babylonjs = () => (
  <IFrame
    aspect={3 / 2}
    url={BABYLONJS_URL}
    title={"babylonjs"}
    canFullscreen
  />
);
const Breakout = () => (
  <IFrame
    aspect={3 / 2}
    mobileAspect={2 / 3}
    url={BREAKOUT_URL}
    title={"breakout"}
    canFullscreen
  />
);
const Platformer = () => (
  <IFrame
    aspect={16 / 10}
    url={PLATFORMER_URL}
    title={"platformer"}
    canFullscreen
  />
);
const HexTiles = () => (
  <IFrame
    aspect={16 / 10}
    url={HEX_TILES_URL}
    title={"hex tiles"}
    canFullscreen
  />
);
const PlainWebGL = () => (
  <IFrame
    aspect={16 / 10}
    url={PLAIN_WEBGL_URL}
    title={"hex tiles"}
    canFullscreen
  />
);
const BigfootMap = () => (
  <IFrame aspect={16 / 10} url={BIGFOOT_MAP_URL} title={"bigfoot sightings"} />
);

export type RouteDefinition = {
  path: string;
  Component: FC;
  children?: RouteDefinition[];
};

export const ROUTES: RouteDefinition[] = [
  { path: "", Component: Homepage },
  {
    path: "projects",
    Component: Projects,
    children: [
      {
        path: "figrid",
        Component: ProjectsFigrid,
      },
      {
        path: "pixijs",
        Component: Pixijs,
        children: [{ path: ":id", Component: Pixijs }],
      },
      { path: "babylonjs", Component: Babylonjs },
      { path: "breakout", Component: Breakout },
      { path: "platformer", Component: Platformer },
      { path: "hex_tiles", Component: HexTiles },
      { path: "webgl", Component: PlainWebGL },
      { path: "bigfoot", Component: BigfootMap },
      {
        path: "shaders",
        Component: Shaders,
        children: [{ path: ":category", Component: Shaders }],
      },
      {
        path: "sketches",
        Component: Sketches,
        children: [{ path: ":id", Component: Sketches }],
      },
    ],
  },
  { path: "photos", Component: Photos },
  // {
  //   path: "posts",
  //   Component: Posts,
  //   children: [{ path: ":id", Component: PostsContent }],
  // },
];

export const Routing = () => {
  const { pathname } = useLocation();

  useEffect(() => {
    const baseRoute = pathname.split("/")[1];
    document.title = baseRoute
      ? `${DOCUMENT_TITLE_ROOT} â€” ${baseRoute}`
      : DOCUMENT_TITLE_ROOT;
  }, [pathname]);

  const renderRoutes = (routes: RouteDefinition[], prefix = "") =>
    routes.reduce((renderedRoutes, { path, Component, children }) => {
      const fullPath = `${prefix}/${path}`;

      if (children) {
        renderedRoutes.push(...renderRoutes(children, fullPath));
      }

      return renderedRoutes.concat(
        <Route path={fullPath} element={<Component />} />
      );
    }, [] as ReactNode[]);

  return (
    <Suspense fallback={<Loader />}>
      <Routes>{renderRoutes(ROUTES)}</Routes>
    </Suspense>
  );
};
