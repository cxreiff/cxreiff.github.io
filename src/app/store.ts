import { createStore, applyMiddleware, compose } from 'redux'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { connectRouter, routerMiddleware } from 'connected-react-router'

import history from './history'
import homepage from '~/src/features/homepage/homepage.slice'

const router = connectRouter(history)

const middleware = [
    ...getDefaultMiddleware(),
    routerMiddleware(history),
];
 
if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger');
  middleware.push(logger);
}

const store = configureStore({
    reducer: {
        router,
        homepage,
    },
    middleware,
})

export type AppDispatch = typeof store.dispatch
export const useAppDispatch = () => useDispatch<AppDispatch>()

export type RootState = ReturnType<typeof store.getState>
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

export default store

