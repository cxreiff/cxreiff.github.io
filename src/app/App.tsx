import React from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'

import 'normalize.css'

import store from './store'
import history from './history'

import Layout from '~/src/features/layout/layout.component'

const App: React.FC = () => (
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <Layout/>
        </ConnectedRouter>
    </Provider>
)

export default App
