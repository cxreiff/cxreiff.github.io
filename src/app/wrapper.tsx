import React, { ComponentType, ReactElement } from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'

import store from './store'
import history from './history'

const wrapper = (Component: ComponentType, props = {}): ReactElement => {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Component {...props} />
      </ConnectedRouter>
    </Provider>
  )
}

export default wrapper
