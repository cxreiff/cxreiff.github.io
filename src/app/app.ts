import { FC } from 'react'
import 'normalize.css'

import wrapper from './wrapper'
import Layout from '~/src/features/layout/layout.component'

const App: FC = () => wrapper(Layout)

export default App
