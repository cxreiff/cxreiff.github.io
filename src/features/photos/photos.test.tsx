import { mount, ReactWrapper } from 'enzyme'

import { Wrapper } from '~/src/app/wrapper'
import Photos from './photos.component'

describe('photos', () => {

    let component: ReactWrapper<typeof Wrapper, void>
    
    beforeAll(() => {
        component = mount<typeof Wrapper, void>(
            <Wrapper>
                <Photos />
            </Wrapper>
        )
    })
    
    afterAll(() => {
        component.unmount()
    })

    it('should render', () => {
        expect(component.find(Photos)).toMatchSnapshot()
    })
})
