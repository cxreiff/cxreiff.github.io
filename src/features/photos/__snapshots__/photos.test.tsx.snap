// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`photos should render 1`] = `
<Photos>
  <section
    className="photos"
  />
</Photos>
`;
